cdc/generation.cc

```diff
@@ -263,6 +263,26 @@ bool should_propose_first_generation(const gms::inet_address& me, const gms::gos
     });
 }
 
+bool is_cdc_generation_optimal(const cdc::topology_description& gen, const locator::token_metadata& tm) {
+    if (tm.sorted_tokens().size() != gen.entries().size()) {
+        // We probably have garbage streams from old generations
+        cdc_log.info("Generation size does not match the token ring, regenerating");
```

Comment:
Printing these messages here assumes that the `is_cdc_generation_optimal` function (which is otherwise a pure function) is called only in a certain context (i.e. if it returns `true`, we'll regenerate). This is an abstraction leak.

But it's such a minor problem, I'll look over it. But in the future please take care of such things.

Reply:
I removed "regenerating" from these logs and added a new INFO log informing about regenerating after a call to `is_cdc_generation_optimal`.